android/app/src/main/java/org/uwaterloo/subletr/pages/createlisting/CreateListingPageView.kt

```diff
@@ -606,74 +563,240 @@ fun LeaseDatePicker(state: DateRangePickerState, onClick: () -> Unit) {
+	bottomSheetState: SheetState,
+	onDismissRequest: () -> Unit,
+	onTakePhotoClick: () -> Unit,
+	onChooseImageClick: () -> Unit,
+	) {
+	ModalBottomSheet(
+		modifier = Modifier,
+		onDismissRequest = onDismissRequest,
+		sheetState = bottomSheetState,
+		containerColor = Color.White,
+		dragHandle = { BottomSheetDefaults.DragHandle(
+			width = dimensionResource(id = R.dimen.xl)
+		)},
+		content = {
+			Box(
+				modifier = Modifier
+					.fillMaxWidth()
+					.height(dimensionResource(id = R.dimen.xxxxxxxl))
+					.background(Color.White)
+			) {
+				Row(
+					modifier = Modifier
+						.fillMaxWidth()
+						.fillMaxHeight()
+						.padding(horizontal = dimensionResource(id = R.dimen.l)),
+					horizontalArrangement = Arrangement.SpaceBetween,
+					verticalAlignment = Alignment.CenterVertically,
+				) {
+					ImageUploadMethodButton(
+						R.drawable.photo_camera_outline_black_64,
+						stringResource(id = R.string.take_photo),
+						onClick = onTakePhotoClick,
+					)
 
-		imageUri?.let {
-			if (Build.VERSION.SDK_INT < 28) {
-				bitmap.value = MediaStore.Images
-					.Media.getBitmap(context.contentResolver,it)
+					ImageUploadMethodButton(
+						R.drawable.photo_outline_black_64,
+						stringResource(id = R.string.choose_image),
+						onClick = onChooseImageClick,
+					)
+				}
+			}
+		},
+	)
+}
+
+@OptIn(ExperimentalMaterial3Api::class)
+@Composable
+fun UploadImages(
+	viewModel: CreateListingPageViewModel,
+	uiState: CreateListingPageUiState.Loaded,
+) {
+	val context = LocalContext.current
+	val coroutineScope = rememberCoroutineScope()
+
+	var imageUris by remember { mutableStateOf<List<Uri?>>(ArrayList()) }
+	var hasImage by remember { mutableStateOf(false) }
+
+	var openImagePicker by rememberSaveable { mutableStateOf(false) }
+	val imagePickerBottomSheetState = rememberModalBottomSheetState(skipPartiallyExpanded = true)
+
+	val imageSelectorLauncher = rememberLauncherForActivityResult(
+		contract = ActivityResultContracts.GetMultipleContents()) {
+		imageUris = it
+		hasImage = true
+	}
+
+	val cameraLauncher = rememberLauncherForActivityResult(
+		contract = ActivityResultContracts.TakePicture(),
+	) {
+		hasImage = it
+	}
 
-			} else {
-				val source = ImageDecoder
-					.createSource(context.contentResolver,it)
-				bitmap.value = ImageDecoder.decodeBitmap(source)
+	LazyRow(
+		modifier = Modifier.fillMaxWidth(),
+		verticalAlignment = Alignment.CenterVertically,
+	) {
+		item {
+			Button(
+				modifier = Modifier
+					.size(dimensionResource(id = R.dimen.xxxxxxl))
+					.border(
+						dimensionResource(id = R.dimen.xxxs),
+						textFieldBorderColor,
+						RoundedCornerShape(dimensionResource(id = R.dimen.s))
+					),
+				shape = RoundedCornerShape(dimensionResource(id = R.dimen.s)),
+				onClick = {
+					coroutineScope.launch {
+						openImagePicker = true
+					}
+				},
+				colors = ButtonDefaults.buttonColors(
+					containerColor = textFieldBackgroundColor,
+					contentColor = Color(0xFF808080),
```

Comment:
Use or make dark mode conscious color from the color constants stuff?
Although I don't think this value actually changes anythings....